trio/_core/_run.py

```diff
@@ -245,6 +291,25 @@ def effective_deadline(self):
         return min(self._scope.deadline, self._parent.effective_deadline())
 
 
+MISNESTING_ADVICE = """Typically this is caused by one of the following:
```

Comment:
Let's start with a nice visible "This is probably a bug in your code, that has caused Trio's internal state to become corrupted. We'll do our best to recover, but from now on there are no guarantees. [blank line] Typically ..."